test/test_foreach.py

```diff
@@ -911,8 +911,51 @@ def test_outplace_with_invalid_grads(self, device, dtype, op):
         self.assertIsNotNone(sample.input[0].grad)
         self.assertIsNone(sample.input[1].grad)
 
+    @ops(
+        filter(
+            lambda op: op.supports_forward_ad,
+            foreach_unary_op_db + foreach_binary_op_db + foreach_pointwise_op_db + foreach_lerp_op_db,
+        ),
+        dtypes=(torch.float,),
+    )
+    def test_forward_mode_AD(self, device, dtype, op):
+
+        patterns_to_tolerate = "|".join([
```

Comment:
I wasn't clear enough but these "expected failures" happen in the actual forward computation with certain Scalar/ScalarList right hand side arguments. `op.sample_inputs` however yields various Scalars and ScalarLists whose most patterns are okay. Thus, I find this fair (not ideal ever though)

Reply:
Okay that makes sense, thanks for the clarification. Also whoops I forgot that you can't skip samples at the OpInof level (although it would be cool if we could).